guide/src/ecosystem/async-await.md

```diff
@@ -268,3 +269,191 @@ async fn main() -> pyo3::PyResult<()> {
     pyo3_asyncio::testing::main().await
 }
 ```
+
+
+# Calling async Python functions in Rust code
+
+In the previous chapter, we learnt how to call Python functions in Rust code. Now, we will look at two ways to call Python async functions from Rust code.
+In python, even an async function is treated like a regular function until called. When called, an async python function is treated as a coroutine.(More on this below.)
+We can either pass it either as a coroutine or either as a regular function.
+
+
+## Async Functions
+
+Before we start, let us take a look at the following snippet.
+
+```python
+async def fx():
+    return "hello, world"
+```
+
+Above, we have defined an async function. We will now execute it in rust code.
+
+In python, even an async function is treated like a regular function until called.
+e.g.
+
+```python
+>>> type(fx)
+<class 'function'>
+>>> type(fx())
+<class 'coroutine'>
+```
+
+We will now write rust functions to execute the python code in rust. The `PyAny` binding will be used to call Python functions in Rust code.
+
+Now, let us take a look at how will we share the Python function in our rust code.
+
+```rust
+
+use pyo3::prelude::*;
+
+#[pyclass]
+struct SomeObject {}
+
+#[pymethods]
+impl SomeObject {
+    #[new]
+    fn new() -> Self {
+        Self {}
+    }
+
+    fn start(&self, python_function: &PyAny) {
+        // where python_function is the function added from python
+    }
+}
+
+#[pymodule]
+pub fn some_module(py: Python<'_>, m: &PyModule) -> PyResult<()> {
+    m.add_class::<Server>()?;
+    pyo3_asyncio::try_init(py);
+    Ok(())
+}
+
+```
+
+Now, to start it in python, we will do something like the following code:
+
+```python
+
+import some_module
+
+some_object = some_module.SomeObject()
+
+async def fx():
+    return "hello world"
+
+some_object.start() # what we pass here, will determine the code we write in our rust start
+```
+
+The `start` function shows how we will be able to send our Python code to rust.
+
+Now, we have two choices. Either we can pass the Python function as the following:
+
+```python
+some_object.start(fx)
+```
+
+or we can pass an awaitable coroutine
+
+```python
+some_object.start(fx())
+```
```

Comment:
I think that introducing this choice already is confusing. Let's just focus on how to pass an awaitable coroutine for now, and use a new section at the end to talk about passing the function (I've suggested calling it "reusing a coroutine").

```suggestion
The following Python code shows how to pass a Python coroutine to the above Rust code:

```python
import some_module

async def fx():
    return "hello world"

some_object.await_in_rust(fx())
```
```